### PR TITLE
fix(NcDateTimePicker): change styling to look more like the native picker

### PR DESCRIPTION
**Pull request by GretaD**
### ☑️ Resolves

- Fix # https://github.com/nextcloud/calendar/pull/6322

### 🖼️ Screenshots

🏚️ Before | 🏡 After
after
Screenshot from 2024-09-30 11-52-42

### 🚧 Tasks

- [ ] ...

### 🏁 Checklist

- [ ] ⛑️ Tests are included or are not applicable
- [ ] 📘 Component documentation has been extended, updated or is not applicable
- [ ] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

### CODE REVIEW

File: src/components/NcDateTimePicker/NcDateTimePicker.vue [hunk 1]
```diff
@@ -477,10 +477,53 @@ export default {
...
+.mx-datepicker-main .mx-table .cell.disabled {
+	border-radius: var(--border-radius-small);
+	background-color: var(--color-background-dark);
+	color: var(--color-text-maxcontrast);
+	opacity: 1;
+}
+
+.mx-datepicker-main .mx-table .cell {
+	border-radius: var(--border-radius-small);
+	min-height: 30px !important;
```
Comment (susnux) [inline src/components/NcDateTimePicker/NcDateTimePicker.vue:499]: This should be probably either default clickable or clickable-small right?

Comment (GretaD) [reply src/components/NcDateTimePicker/NcDateTimePicker.vue:499]: yes, and also, i need to fix the classes to be shown as BEM.

Comment (susnux) [reply src/components/NcDateTimePicker/NcDateTimePicker.vue:499]: But that are internal classes, so no renaming to BEM possible, no?

Comment (GretaD) [reply src/components/NcDateTimePicker/NcDateTimePicker.vue:499]: i meant:

```
main {
.cell {
.today {
.other.. {
}
 }
  }
   }
```

**Review by nimishavijay — CHANGES_REQUESTED** [review]
Looks really good! Very big improvement when we compare it to the current design :) 
- [ ] Is it possible to hide the first row in this case? The topmost week that is shown should be the week containing the 1st of the month (eg. we are on September but the last week of August is also shown)
- [x] Is it possible to hide the double arrows which skip to the same month next year (`«` and `»`) since I imagine that functionality is not used very often. This would also provide more horizontal space for the month name :) 
- [x] As @susnux mentioned the size should be a variable (`--clickable-area-small` should work here)
- [x] Nice-to-have: Show full month name

**Comment by GretaD** [comment]
> Looks really good! Very big improvement when we compare it to the current design :)
> 
> * [ ]  Is it possible to hide the first row in this case? The topmost week that is shown should be the week containing the 1st of the month (eg. we are on September but the last week of August is also shown)

Unfortunately, i cannot find a way, but i removed the background color, so it doesnt look too crowded.

The rest is done. I updated to screenshot

**Review by susnux — CHANGES_REQUESTED** [review]
Looks nice, **but** it is no longer possible to select a year that is more in the past / future as the buttons are removed:

before | now
---|---
image|image


Also when selecting the month the year should be centered like before:

before | now
---|---
image|image

**Comment by GretaD** [comment]
> Looks nice, **but** it is no longer possible to select a year that is more in the past / future as the buttons are removed:


ahhh thanks for testing it, @nimishavijay i guess we need the buttons :D

**Comment by GretaD** [comment]
> Please also have a look at #6125. I missed that you were also working on the date picker design 🙈
> 
> Ref [nextcloud/calendar#6324](https://github.com/nextcloud/calendar/issues/6324)

we can merge them, no problem, the change is the same, with small difference, like the hover button state. Do you want to include mine at yours, or should i add yours here?

**Review by susnux — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**